src/App.js

```diff
@@ -22,6 +23,7 @@ function App() {
         <Route path="/hacks" component={HacksPage} />
         <Route exact path="/hack/:id" component={HackDetails} />
         <Route path="/hack/:id/edit" component={EditHack} />
+        <Route path="/hack/:id/submit" component={SubmitHack} />
```

Comment:
Honestly, its tempting to just use the component prop on react-router route, but i feel like unless the component actually needs to be router aware, you're better off using the render prop and passing in the params as props... this will save you in testing effort (component doesn't know anything about routes) and also for reusability if you ever want to use the component not in a routing context.

```
<Route path="/hack/:id/submit" render={({ match: { params: id }}) => <SubmitHack hackId={id} />} />
```